### PR TITLE
legacy/1.x: Backport #909 to use global.json to pin .Net Core SDK version and update .Net SDK version to be the same as in master (2.1.801)

### PR DESCRIPTION
**Pull request by bergmeister**
Backport #909 to use global.json to pin .Net Core SDK version and update .Net SDK version to be the same as in master.
This makes the 2 branches not only more consistent but makes building locally easier (I could not build the legacy branch at first yesterday because I have the .Net Core 3 SDK installed)

### CODE REVIEW

File: PowerShellEditorServices.build.ps1 [hunk 1]
```diff
@@ -34,7 +34,7 @@ if ($PSVersionTable.PSEdition -ne "Core") {
 
 task SetupDotNet -Before Clean, Build, TestHost, TestServer, TestProtocol, TestPowerShellApi {
 
-    $requiredSdkVersion = "2.0.0"
```
Comment (bergmeister) [inline PowerShellEditorServices.build.ps1]: Note that 2.0 has already reached the end of its life a year ago!
https://dotnet.microsoft.com/download/dotnet-core

**Comment by bergmeister** [comment]
It seems changing the SDK from 2.0 to 2.1 breaks the Windows builds in different ways

**Comment by bergmeister** [comment]
@poshchan please retry

**Comment by PoshChan** [comment]
@bergmeister, I do not understand: please retry

<details>
<summary>Commands available in this repo for you:
</summary>
<ul><li>
<code>get failures</code> this will attempt to get the latest failures for all of the target pipelines
</li>
<li>
<code>remind me in &lt;value&gt; &lt;units&gt;</code> this will create a reminder that will be posted after the specified duration &lt;value&gt; is a number, and &lt;units&gt; can be <code>minutes</code>, <code>hours</code>, or <code>days</code> (singular or plural)
</li>

</ul>
</details>

**Comment by TylerLeonhardt** [comment]
@bergmeister I usually just use the checks tab

**Comment by TylerLeonhardt** [comment]
The legacy branch is an odd one. It targets net451 and netstandard1.3 I believe. It might not be worth making this change.

**Comment by TylerLeonhardt** [comment]
#1103 has a lot of this code change in it. Closing this.

**Pull request closed without merging**